pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +271,86 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+
+	partitionList, err := t.GetPartitionsFromTimeRange(txn, startTime, endTime)
+	stats.PartitionCount = len(partitionList)
+	if err != nil {
+		return resources, stats, errors.Wrapf(err, "failed to get partitions from table:%v, from startTime:%v, to endTime:%v", t.tableName, startTime, endTime)
+	}
+
+	tablePrefix := "/" + t.tableName + "/"
+	for _, currentPartition := range partitionList {
+		var seekStr string
+
+		// when keyPrefix does not have partition info(which implies it may not contain kind,namespace,and etc), we seek from /tableName/currentPartition/
+		if keyPrefix.PartitionId == "" {
+			seekStr = tablePrefix + currentPartition + "/"
+		} else {
+			// update keyPrefix with current partition
+			keyPrefix.SetPartitionId(currentPartition)
+			seekStr = keyPrefix.String()
+		}
+
+		itr := txn.NewIterator(badger.IteratorOptions{Prefix: []byte(seekStr)})
+		defer itr.Close()
+
+		//in worst case, when seekStr = /table/partition, we need to iterate a key list and return all of them
+		//in most cases, we should only hit one result per partition
+		for itr.Seek([]byte(seekStr)); itr.ValidForPrefix([]byte(seekStr)); itr.Next() {
+			stats.RowsVisitedCount += 1
+			key := EventCountKey{}
+			err := key.Parse(string(itr.Item().Key()))
```

Comment:
I was expecting us to copy the key prefix and override the partition.  This seems like it would lose the name/namespace/kind for example

Reply:
the override is already happen in line 298,right? it updates its partition with current partition, and move the pointer to the new seek string, did I miss anything?

Reply:
Why do we need to parse a key if we already have one?

Reply:
Sorry, this was my mistake.  I mis-read the code.  Ignore all my comments about this